From dd2bba8803b521c6322862f2cb6c859d4d2ba684 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:10 -0400
Subject: [tools] perf/traceevent: Check for spaces in character array

Message-id: <1408462094-14194-262-git-send-email-jolsa@redhat.com>
Patchwork-id: 88283
O-Subject: [PATCH RHEL7.1 BZ1131394 261/325] tools lib traceevent: Check for spaces in character array
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 5efb9fbd5f1bfe4435bd0a3ea5f0e187875509c2
Author: Steven Rostedt (Red Hat) <rostedt@goodmis.org>
Date: Fri Nov 1 17:53:58 2013 -0400

description
===========
Currently when using the raw format for fields, when looking at a
character array, to determine if it is a string or not, we make sure all
characters are "isprint()". If not, then we consider it a numeric array,
and print the hex numbers of the characters instead.

But it seems that '\n' fails the isprint() check! Add isspace() to the
check as well, such that if all characters pass isprint() or isspace()
it will assume the character array is a string.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/event-parse.c b/tools/lib/traceevent/event-parse.c
index e1c743c..85cbbdd 100644
--- a/tools/lib/traceevent/event-parse.c
+++ b/tools/lib/traceevent/event-parse.c
@@ -3981,7 +3981,7 @@ static int is_printable_array(char *p, unsigned int len)
  unsigned int i;
 
  for (i = 0; i < len && p[i]; i++)
-  if (!isprint(p[i]))
+  if (!isprint(p[i]) && !isspace(p[i]))
       return 0;
  return 1;
 }
-- 
1.7.1